src/eth/storage/stratus_storage.rs

```diff
@@ -52,7 +60,11 @@ impl StratusStorage {
 
     /// Creates a new storage with the specified temporary and permanent implementations.
     pub fn new(temp: Box<dyn TemporaryStorage>, perm: Box<dyn PermanentStorage>) -> Self {
-        Self { temp, perm }
+        Self {
+            temp,
+            perm,
+            last_saved_block_number: AtomicU64::new(0),
```

Comment:
I think this attribute is not necessary.

We already have the necessary values being set in `self.perm.read_mined_block_number()` and `self.temp.read_pending_block_number()`.

We should check the invariants based on these values.